File: game/restaurants/restaurants.go
package restaurants

type Restaurant struct {
	name      string
	cash      float32
	inventory Inventory
	menu      Menu
}

func (rest *Restaurant) giveMoney(money float32) {
	rest.cash += money
}

func (rest *Restaurant) cook(ramen Ramen) {
	price := rest.menu.getPrice(ramen)
	rest.inventory.withdraw(ramen.receipe)
	rest.giveMoney(price)
}

File: game/src/restaurants/restaurants.rs
use super::Menu;
use super::Inventory;

use super::Order;
use super::OrderQueue;

pub type RestaurantKey = i32;
pub struct Restaurant {
    pub id: RestaurantKey,
    pub name: String,
    pub cash: f64,
    pub menu: Menu,
    pub stocks: Inventory,
    placed_orders: OrderQueue,
}

impl Restaurant {
    pub fn default(name: &str) -> Restaurant {
        Self {
            id: 0,
            name: String::from(name),
            cash: 0.,
            menu: Menu::default(),
            stocks: Inventory::new(),
            placed_orders: OrderQueue::new(),
        }
    }

    pub fn new(name: &str, cash: f64, menu: Menu, stocks: Inventory) -> Restaurant {
        Self {
            id: 0,
            name: String::from(name),
            cash: cash,
            menu: menu,
            stocks: stocks,
            placed_orders: OrderQueue::new(),
        }
    }

    pub fn place_order(&mut self, order: Order) {
        self.placed_orders.place_order(order);
    }
}